### PR TITLE
refactor: when deleting a user, delete the users keys too

### LINKED ISSUE #2398: ssh key not removed when delete user
# Describe the bug

Deleting a lagoon user should also remove ssh keys.
Now the deletion of a user doesn't touch ssh keys that remain in left-over status.

## To Reproduce

1. Query user and retrieve info about ssh keys
2. Query user by `UserBySshKey` GraphQL query
3. Delete the user via GraphQL or via lagoon-cli
4. Query user by `UserBySshKey` GraphQL query and get the message `"message": "User not found: <id_of_the_user>"`
5. Query the table `ssh_key` of api-db database and check the ssh key is still there 

## Expected behavior

Deletion of ssh keys belonging to the deleted user

### PR DESCRIPTION
**Pull request by shreddedbacon**
# Checklist

- [ ] Affected Issues have been mentioned in the Closing issues section
- [ ] Documentation has been written/updated
- [ ] PR title is ready for inclusion in changelog

Currently if a user is deleted, their SSH keys remain in the database.

For example, if we have John Smith and they have an existing account with `jsmith@example.com` and their single ssh key attached to the user, but Johns account is accidentally deleted. A new account is created, but when John attempts to re-add the key, it fails because the key is already in the database linked to the previous account.

# Closing issues

closes #2398

### CODE REVIEW

File: services/api/src/resources/user/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { Pool } from 'mariadb';
+import { query } from '../../util/db';
+import { Sql } from './sql';
+// import { logger } from '../../loggers/logger';
+
+export const Helpers = (sqlClientPool: Pool) => {
```
Comment (rocketeerbkw) [inline services/api/src/resources/user/helpers.ts]: The models were created to supersede the helper files pattern. All this should be moved into to the User model (part of, or called by, `UserModel.deleteUser`) which should be the single place for business logic.

File: services/api/src/resources/user/sql.ts [hunk 2]
```diff
@@ -14,4 +14,10 @@ export default {
       .andWhere('sk.key_type', keyType)
       .select('user_ssh_key.usid')
       .toString(),
+  deleteUserSshKeys: (id: string) =>
```
Comment (rocketeerbkw) [inline services/api/src/resources/user/sql.ts]: I think this leaves behind records in the `user_ssh_key` table?

**Review by rocketeerbkw — CHANGES_REQUESTED** [review]

File: services/api/src/resources/user/sql.ts (hunk 2 above)
Comment (shreddedbacon) [reply services/api/src/resources/user/sql.ts]: ah yep

File: services/api/src/resources/user/helpers.ts (hunk 1 above)
Comment (shreddedbacon) [reply services/api/src/resources/user/helpers.ts]: rgr

**Review by rocketeerbkw — APPROVED** [review]

**Pull request merged**